File: BackEnd/Domain/Aggregates/Ticket.cs
using SharedKernel.Aggregate;

namespace Domain.Aggregates;

public class Ticket : AggregateRoot<Guid>
{
    public Guid ProjectId { get; set; }
    public Guid TicketStatusId { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public ICollection<Comment> Comments { get; set; }
    public Project Project { get; set; }
    public TicketStatus TicketStatus { get; set; }
    public ICollection<User> Users { get; set; }
    
    public static Ticket Create(Guid projectId, Guid ticketStatusId, string title, string description, List<User> users)
    {
        return new Ticket
        {
            Id = Guid.CreateVersion7(),
            ProjectId = projectId,
            TicketStatusId = ticketStatusId,
            Title = title,
            Description = description,
            Users = users
        };
    }
    public void Update(Guid ticketStatusId, string title, string description, List<User> users)
    {
        TicketStatusId = ticketStatusId;
        Title = title;
        Description = description;
        Users = users;
    }
    
}

File: flutter/lib/domain/aggregates/ticket/ticket.dart
import 'package:dart_mappable/dart_mappable.dart';
import 'package:enterprise_management/domain/aggregates/comment/comment.dart';
import 'package:enterprise_management/domain/aggregates/user/user.dart';
import 'package:enterprise_management/shared_kernel/aggregate/aggregate_root.dart';

part 'ticket.mapper.dart';
@MappableClass()
class Ticket extends AggregateRoot<String> with TicketMappable {
  Ticket({required super.id, required this.title, required this.description, required this.comments, required this.projectId, required this.ticketStatusId, required this.users});

  final String title;
  final String description;
  final List<Comment> comments;
  final String projectId;
  final String ticketStatusId;
  final List<User> users;
}